archunit/src/test/java/com/tngtech/archunit/ArchConfigurationTest.java

```diff
@@ -208,6 +214,49 @@ public void allows_to_specify_custom_properties() {
         configuration.getProperty("not.there");
     }
 
+    @Test
+    public void allows_to_override_any_property_via_system_property() {
+        String customPropertyName = "my.custom.property";
+        String otherPropertyName = "my.other.property";
+        writeProperties(
+                ArchConfiguration.ENABLE_MD5_IN_CLASS_SOURCES, false,
+                customPropertyName, "original",
+                otherPropertyName, "other"
+        );
+
+        ArchConfiguration configuration = testConfiguration(PROPERTIES_FILE_NAME);
+
+        assertThat(configuration.md5InClassSourcesEnabled()).as("MD5 sum in class sources enabled").isFalse();
+        assertThat(configuration.getProperty(customPropertyName)).as("custom property").isEqualTo("original");
+
+        System.setProperty("archunit." + ArchConfiguration.ENABLE_MD5_IN_CLASS_SOURCES, "true");
+        System.setProperty("archunit." + customPropertyName, "changed");
+
+        assertThat(configuration.md5InClassSourcesEnabled()).as("MD5 sum in class sources enabled").isTrue();
+        assertThat(configuration.getProperty(customPropertyName)).as("custom property").isEqualTo("changed");
+        assertThat(configuration.getSubProperties(subPropertyKeyOf(customPropertyName))).containsExactly(
+                entry(subPropertyNameOf(customPropertyName), "changed"),
+                entry(subPropertyNameOf(otherPropertyName), "other"));
+    }
+
+    private String subPropertyKeyOf(String customPropertyName) {
+        return customPropertyName.split("\\.")[0];
+    }
+
+    private String subPropertyNameOf(String customPropertyName) {
+        return customPropertyName.split("\\.", 2)[1];
+    }
+
+    private void writeProperties(Object... props) {
+        checkArgument(props.length % 2 == 0, "There are more keys than values inside of %s", Arrays.toString(props));
+        ImmutableMap.Builder<String, Object> map = ImmutableMap.builder();
+        for (int i = 0; i < props.length; i += 2) {
+            checkArgument(props[i] instanceof String, "Array entry %s is supposed to be a property name, but is no string", props[i]);
+            map.put((String) props[i], props[i + 1]);
+        }
+        writeProperties(map.build());
+    }
+
     private void writeProperties(Map<String, ?> props) {
```

Comment:
If `empty_property_file` used `writeProperties()`, this could be inlined to `writeProperties(Object...)`, which in turn wouldn't have to build the `Map` anymore.